crates/oci-distribution/src/client.rs

```diff
@@ -751,15 +752,24 @@ impl ClientProtocol {
 }
 
 /// A token granted during the OAuth2-like workflow for OCI registries.
-#[derive(serde::Deserialize, Default)]
-struct RegistryToken {
-    #[serde(alias = "access_token")]
-    token: String,
+#[derive(Deserialize)]
+#[serde(untagged)]
+#[serde(rename_all = "snake_case")]
+enum RegistryToken {
+    Token { token: String },
```

Comment:
Note: I believe some registries send _both_ fields. I originally tried 

```rust
#[derive(Deserialize)]
#[serde(rename_all = "snake_case")]
enum RegistryToken {
    Token(String),
    AccessToken(String)
}
```

but this would cause an error if both fields are present. The `untagged` approach that I use here simply selects the first match without error.

Reply:
It's an odd case but not unheard of, yes.